jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/FusekiServer.java

```diff
@@ -1021,6 +1021,7 @@ private void servletsAndFilters(ServletContextHandler context) {
                 addServlet(context, "/$/stats/*", new ActionStats());
             if ( withMetrics )
                 addServlet(context, "/$/metrics", new ActionMetrics());
+            // TODO Should we support registering other functionality e.g. /$/backup/* and /$/compact/*
```

Comment:
Not sure this is possible because `jena-fuseki-main` does not depend on `jena-fuseki-webapp` from which the backup and compact functionality originate.

Reply:
There are a couple of big issues here.

1. Security. So far fuseki-main additional operations under `/$/`are "mostly harmless" but this is a step more impactful.
1. The general area of admin for Fuseki/main where my concern is that we are building in a URL pattern and _maybe_ a better design is add it as a service on the `/myTDB2dataset/compact`.

Suggestion: add this admin function now for fuseki-webapp and fuseki-main later.

Let's discuss on the JIRA ticket as it is not about the jena-webapps code which looks fine.